harness/determined/launch/deepspeed.py

```diff
@@ -100,29 +104,42 @@ def create_deepspeed_env_file() -> None:
         environ = os.environ.copy()
         for k, v in environ.items():
             if k in INCLUDE:
-                f.write(f"{k}={v}\n")
+                if deepspeed_version >= version.parse("0.6.2"):
+                    f.write(f"{k}={shlex.quote(v)}\n")
+                else:
+                    f.write(f"{k}={v}\n")
 
 
-def create_run_command(master_address: str, hostfile_path: str) -> List[str]:
+def create_run_command(master_address: str, hostfile_path: Optional[str]) -> List[str]:
     # Construct the deepspeed command.
-    deepspeed_process_cmd = [
-        "deepspeed",
-        "-H",
-        hostfile_path,
-        "--master_addr",
-        master_address,
-        "--no_python",
-        "--no_local_rank",
-        "--",
-    ]
+    deepspeed_process_cmd = ["deepspeed"]
+    if hostfile_path is not None:
+        deepspeed_process_cmd += ["-H", hostfile_path]
+    deepspeed_process_cmd += ["--master_addr", master_address, "--no_python", "--no_local_rank"]
+    if deepspeed_version > version.parse("0.6.4"):
+        deepspeed_process_cmd.append("--no_ssh_check")
```

Comment:
question: how did we avoid the need for this flag before?

Reply:
one of the PRs linked above recently added ssh check to the deepspeed launcher but it doesn't have all the flags we require for ssh (i.e. StrictHostKeyChecking=no)